core/src/test/scala/eu/shiftforward/apso/akka/http/ProxySupportSpec.scala

```diff
@@ -153,6 +156,14 @@ class ProxySupportSpec(implicit ee: ExecutionEnv) extends Specification with Spe
         status == OK
         responseAs[String] must be_==("/remote-proxy")
       }
+
+      proxy.sendRequest(Get("/proxied"), failOnDrop = false).flatMap {
```

Comment:
Can you add tests here for the case where `failOnDrop = true` and for the cases where the request is effectively dropped? Creating a proxy with a `reqQueueSize` of zero should do the trick.